terraformprovider/resource_object_group.go

```diff
@@ -133,9 +151,36 @@ func resourceObjectGroup() *schema.Resource {
 				Default:  "",
 			},
 		},
+		CustomizeDiff: resourceObjectGroupCustomizeDiff,
 	}
 }
 
+func resourceObjectGroupCustomizeDiff(ctx context.Context, diff *schema.ResourceDiff, meta interface{}) error {
```

Comment:
I'm not sure I fully understand the CustomizeDiff behaviour 🤔 
From what I read in the docs https://www.terraform.io/plugin/sdkv2/resources/customizing-differences it looks like this function should return bool, isn't it? Once again, I never used it and not sure I understand it correctly...
The way I see it now - it looks more like a validation and I would expect it to be called smth like `validate....`, but I may be wrong.

Reply:
The purpose of this function is to modify the diff generated by terraform, but people use it for validation, because there is no other way to access several fields at the same time. Same here - the field type=regex requires "patterns" to be defined and type=whitelist requires "includes".